src/uct/api/uct_def.h

```diff
@@ -137,6 +144,66 @@ typedef struct uct_iov {
 } uct_iov_t;
 
 
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Remote data attributes field mask.
+ *
+ * The enumeration allows specifying which fields in @ref uct_cm_remote_data are
+ * present, for backward compatibility support.
+ */
+enum uct_cm_remote_data_field {
+    /** Enables @ref uct_cm_remote_data::dev_addr */
+    UCT_CM_REMOTE_DATA_FIELD_DEV_ADDR              = UCS_BIT(0),
+
+    /** Enables @ref uct_cm_remote_data::dev_addr_length */
+    UCT_CM_REMOTE_DATA_FIELD_DEV_ADDR_LENGTH       = UCS_BIT(1),
+
+    /** Enables @ref uct_cm_remote_data::conn_priv_data */
+    UCT_CM_REMOTE_DATA_FIELD_CONN_PRIV_DATA        = UCS_BIT(2),
+
+    /** Enables @ref uct_cm_remote_data::conn_priv_data_length */
+    UCT_CM_REMOTE_DATA_FIELD_CONN_PRIV_DATA_LENGTH = UCS_BIT(3)
+};
+
+
+/**
+ * @ingroup UCT_RESOURCE
+ * @brief Data received from the remote peer.
+ *
+ * The remote peer's device address, the data received from it and their lengths.
+ * Used with the client-server API on a connection manager.
+ */
+typedef struct uct_cm_remote_data {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_cm_remote_data_field. Fields not specified by this mask
+     * will be ignored.
+     */
+    uint64_t                field_mask;
+
+    /**
+     * Device address of the remote peer.
+     */
+    const uct_device_addr_t *dev_addr;
+
+    /**
+     * Length of the remote device address.
+     */
+    size_t                  dev_addr_length;
+
+    /**
+     * Points to the received data. This is the private data that was passed to
+     * @ref uct_ep_params_t::sockaddr_pack_cb.
+     */
+    const void              *conn_priv_data;
+
+    /**
+     * Length of the received data from the peer.
+     */
+    size_t                  conn_priv_data_length;
```

Comment:
is this in or out ?

Reply:
@shamisp this is 'in' - since this a descriptor of the data received from the peer.

Reply:
Line 194: _**Pointer**_ to the the received data.